account_default_draft_move/invoice_view.xml

```diff
@@ -22,7 +22,9 @@
       <field name="type">form</field>
       <field name="arch" type="xml">
         <xpath expr="//button[@name='invoice_cancel']" position="after">
-          <button name="invoice_cancel" states="open" string="Cancel Invoice" groups="account.group_account_invoice"/>
+          <field name="update_posted" invisible="1"/>
+          <button name="invoice_cancel" states="open" string="Cancel Invoice" groups="account.group_account_invoice"
+                  attrs="{'invisible': ['|', ('update_posted', '=', False)]}"/>
```

Comment:
@sbidoul do you test it ?
Because if I correctly remember, it comes from the inheritance and it is put as first parameter, and we want a or with other parameter (I think states parameter). (just check)